packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -86,4 +90,28 @@ class UpgradeCommand extends FlutterCommand {
       || _gitChangedRegex.hasMatch(line)
       || line == 'Fast-forward';
   }
+
+  /// Find and return the project root directory relative to the specified
+  /// directory or the current working directory if none specified.
+  /// Return `null` if the project root could not be found
+  /// or if the project root is inside the flutter repository.
```

Comment:
Fixed. 

There's a comment in [`_commandValidator()`](https://github.com/danrubel/flutter/blob/da77e56cd06f2582aadb3cb7cbcc22746830cd30/packages/flutter_tools/lib/src/runner/flutter_command.dart#L198) along the lines of "don't run this in the root of your git clone of Flutter" which lead me to add that safety check. I've modified it to  allow it to be run in a project contained in the flutter repo (e.g. `examples/flutter_gallery`).